src/mapgen.cpp

```diff
@@ -2690,53 +2691,38 @@ class jmapgen_nested : public jmapgen_piece
         class neighborhood_check
         {
             private:
-                // To speed up the most common case: no checks
-                bool has_any = false;
-                std::array<std::set<oter_str_id>, om_direction::size> neighbors;
-                std::set<oter_str_id> above;
+                std::unordered_map<direction, cata::flat_set<oter_str_id>> neighbors;
```

Comment:
Explicit hash function for direction might be needed to workaround GCC 5.5

Reply:
Done.